devito/operator.py

```diff
@@ -121,89 +122,62 @@ def __init__(self, expressions, **kwargs):
-                if original.is_CompositeData:
-                    for orig, child in zip(original.children, arg.children):
-                        o_vals[orig.name] = child
-
-        # Replace the overridden values with the provided ones
-        for argname in o_vals.keys():
-            arguments[argname] = o_vals[argname]
-
-        # Traverse positional args and infer loop sizes for open dimensions
-        f_args = [(name, f) for name, f in arguments.items()
-                  if isinstance(f, SymbolicData)]
-        for fname, f in f_args:
-            arguments[fname] = self._arg_data(f)
-            shape = self._arg_shape(f)
-
-            # Ensure data dimensions match symbol dimensions
-            for i, dim in enumerate(f.indices):
-                # We don't need to check sizes for buffered dimensions
-                # against data shapes, all we need is the size of the parent.
-                if dim.is_Buffered:
-                    continue
-
-                # Check data sizes for dimensions with a fixed size
-                if dim.size is not None:
-                    if not shape[i] <= dim.size:
-                        error('Size of data argument for %s is greater than the size '
-                              'of dimension %s: %d' % (fname, dim.name, dim.size))
-                        raise InvalidArgument('Wrong data shape encountered')
-                    else:
-                        continue
-
-                if dim_sizes[dim.name] is None:
-                    # We haven't determined the size of this dimension yet,
-                    # try to infer it from the data shape.
-                    dim_sizes[dim.name] = shape[i]
-                else:
-                    # We know the dimension size, check if data shape agrees
-                    if not dim_sizes[dim.name] <= shape[i]:
-                        error('Size of dimension %s was determined to be %d, '
-                              'but data for symbol %s has shape %d.'
-                              % (dim.name, dim_sizes[dim.name], fname, shape[i]))
-                        raise InvalidArgument('Wrong data shape encountered')
-
-        # Make sure we have defined all buffered dimensions and their parents,
-        # even if they are not explicitly given or used.
-        d_args = [d for d in arguments.values() if isinstance(d, Dimension)]
-        for d in d_args:
-            if d.is_Buffered:
-                if dim_sizes[d.parent.name] is None:
-                    dim_sizes[d.parent.name] = dim_sizes[d.name]
-                if dim_sizes[d.name] is None:
-                    dim_sizes[d.name] = dim_sizes[d.parent.name]
+        for x in self.t_args + self.dims + self.s_args:
+            x.reset()
+
+    def arguments(self, *args, **kwargs):
+
+        new_params = {}
+        for k, v in kwargs.items():
+            if isinstance(v, CompositeData):
+                orig_param = [x for x in self.symbolic_data if x.name == k][0]
```

Comment:
They use the same mechanism as the other objects. This section only pulls out the children and adds them to kwargs. The rest of the method treats them the same as any other symbolic data arguments.